mcstatus/tests/test_retry_decorator.py

```diff
@@ -28,13 +30,9 @@ def func():
         elif x == 1:
             raise RuntimeError("Second error")
 
-    try:
+    # We should get the last exception on failure (not OSError)
```

Comment:
Perhaps these comments would be better off as part of the name of the test function?
`def test_sync_failure_raises_only_the_exception_from_the_final_attempt`

Same for the async version too.

Reply:
That's a really long name, I could make them docstrings, I think pytest does show those when that test fails. However I'd probably avoid doing this from here, this PR is about adding flake8, I wanted to make a PR that refactors the testing completely, this can be a part of it but I don't feel like this should be done from here, it has no connection with flake8.

Reply:
You took the time to update the comment when really the test should just have a more descriptive name.

Reply:
I haven't updated the comment, I just got rid of assert False, which was failing `flake8-bugbear` and moved the comment

Reply:
Long names are pretty standard for pytest. I'll do this after merging, since I also want to add a failing test for the failed socket to reproduce AttributeError.